frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -10,12 +13,13 @@ import {
 
 type InstantSearchProviderProps = React.PropsWithChildren<AlgoliaProps>;
 
+type ApiKey = string;
```

Comment:
For the same reason I'd make a constant or an enum instead of repeating the same thing in several places, the better to ensure consistency. I'm just as curious why you wouldn't :-)

Reply:
I agree about being consistent and DRY, but my question is: how is api key type different from any other scalar type in the codebase?

For example, take these props:

```ts
export type AlgoliaProps = {
   url: string;
   indexName: string;
   serverState?: Partial<InstantSearchServerState>;
   apiKey: ApiKey;
};
```

should those be refactored like this?

```ts

type URLType = string;
type IndexName = string;

export type AlgoliaProps = {
   url: URLType;
   indexName: IndexName;
   serverState?: Partial<InstantSearchServerState>;
   apiKey: ApiKey;
};
```

Reply:
That's a good point. Doing that for every scalar would definitely make the code worse. I suppose it might make sense if you couldn't guess the type by the field name. Ok, you convinced me!